spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/mapper/ColumnMapper.scala

```diff
@@ -23,7 +23,7 @@ import scala.reflect.ClassTag
   * }}}
   */
 trait ColumnMapper[T] extends Serializable {
-  def columnMap(tableDef: TableDef): ColumnMap
+  def columnMap(tableDef: TableDef, aliases: Map[String, String] = Map.empty): ColumnMap
```

Comment:
I'm not sure applying aliases to table should be really a concern of a `ColumnMapper`. If we're selecting a subset of columns from a table, why not pass a projection of the original table as a tableDef param, i.e. apply selection + aliases to the original tableDef and then pass it to the column mapper to obtain the column map? This way we wouldn't need to put aliasing logic in all ColumnMappers, and ColumnMappers would be simpler.